include/IECore/MeshAlgo.h

```diff
@@ -54,6 +54,8 @@ std::pair<PrimitiveVariable, PrimitiveVariable> calculateTangents( const MeshPri
 	const std::string &position = "P"
 );
 
+PrimitiveVariable resamplePrimitiveVariable( const MeshPrimitive *mesh, const std::string &primitiveVariable, PrimitiveVariable::Interpolation interpolation );
```

Comment:
Yeah, we've tended to use pointers throughout Cortex/Gaffer where arguably we should have used references would make it clear that NULL is not acceptable. Not sure whether to slowly move to references, change wholesale, or just stick with what we have.

For the `calculateTangents()` thing I think it's fine to stick to the "pass a name" approach for now, because it allows us to make the transition from the separate "s" and "t" variables we have now to a single V2fVectorData "st", without changing the API. Once we've done that we can add a version to take a `PrimitiveVariable &` instead.

Here's another option for avoiding the "matching interpolations" trap :

4. Define `resamplePrimitiveVariable()` so that it modifies the PrimitiveVariable in place. The MeshPrimitive argument could still be const, but the PrimitiveVariable argument would be non-const.

I'm fairly partial to this, as it makes the ResamplePrimitiveVariables node implementation about as simple as it could be :

```
MeshPrimitivePtr result = inMesh->copy();
for( auto x : result->variables )
{
    if( StringAlgo::matchMultiple( x.first, nameMatcher ) )
    {
        resamplePrimitiveVariable( result, x.second, interpolation );
    }
}
```